_getline.c

```diff
@@ -0,0 +1,97 @@
+#include "shell.h"
+
+/**
+ * _getline - read a line from a file descriptor and allocate memory for it.
+ * @fd: the file descriptor from which to read the line.
+ * @line: a pointer to a char pointer where the resulting line will be stored.
+ *
+ * This function reads a line from the specified file descriptor and allocates
+ * memory to store the line. The memory for the line should be freed by the
+ * caller when it is no longer needed.
+ *
+ * Return: Upon success, the function returns the number of characters read,
+ * excluding the null-terminator. On end of file or error, it returns -1.
+ */
+
+int _getline(char **line, const int fd)
+{
+	static char *rd;
+	char buffer[SIZE];
+	int pos;
+
+	pos = (rd != NULL) ? (int)(_strchr(rd, '\n') - rd) : -1;
+
+	if (pos >= 0)
+	{
+		func(*line, &rd, pos);
```

Comment:
is it a typo or did you define a function called `func`? if so try to use a meaningful name.

Reply:
yes i forgot to change the name after writing the code